ui-v2/app/utils/remove-null.js

```diff
@@ -0,0 +1,9 @@
+export default function(obj) {
+  // non-recursive for the moment
+  return Object.keys(obj).reduce(function(prev, item, i, arr) {
+    if (obj[item] !== null) {
+      return { ...prev, ...{ [item]: obj[item] } };
```

Comment:
Making a new shallow clone for every non-null key strikes me as expensive. Have you load-tested this out of curiosity?

Since this is a single function call and prev isn't provided as an argument, you can mutate `prev` without being unsafe.

Reply:
I also just noticed this: `...{ [item]: obj[item] }` which is the same as `[item]: obj[item]` isn't it?

Reply:
Re: expensive
Right now I'm only doing it for an API request that returns a single pojo, and I've specifically not used it elsewhere at the minute. I'd prefer to replace `attr('string')` with something that always gives me a string and use that everywhere.
I hadn't done any load testing, but out of interest I just did. Here's what I did:

1. Created an array of 1,000,000 pojos like `{Name: 'name', Value: null}`.
2. Called `removeNull` with various implementations on every item in the array using a `for` loop, and repeated that 10 times (so 10 x 1,000,000), I did the 10 x repeat in a `map` to gather up the results easily.
3. Took an average of those 10 times
4. Just refreshed a few times to get a 'more or less' ms

I used 3 different implementations for `removeNull`

1. The implementation from above: ~300ms
2. I switched out the `{...prev, ...{}}` etc for `prev[item] = obj[item]`: ~100ms
3. Out of interest I got rid of the `reduce` and just used `delete obj[item]` to mutate instead: ~50ms

So yeah, at 1,000,000 items it shaves 200ms off to use impl. 2 - but on a reasonably good laptop. I had a quick try at increasing the amount of properties on the pojo to 6 props, and got that up to a 1 second difference between impl. 1 and 2 at 1,000,000 items. Interestingly when upping the props, there was next to no difference between impl. 2 and impl. 3

So yeah, right now it just complicates the code without having any impact, and later could definitely be a big 'trippy uppy' later down the line if I end up with millions of props.

👍 

Re: destructuring syntax
Yup the same, my head is still in `Object.assign` 😄 . I'm gonna commit in your other suggestion so that will go from here. If I do change style I'll go through and see if I'm doing it elsewhere (as well as the previous PR) and switch them all at the same time.